### ISSUE TITLE
Constraints on Consul Catalogue not working as expected

### ISSUE THREAD

**Issue by ummecasino** [containous/traefik#703]
Hey guys, 

I'm currently switching from reverse proxying with Nginx/consul-template to Traefik. I'm facing some problems with seperating different environments in Consul by different tags as I used to with consul-template.  
I got 2 environments managed by the same consul instance, so I got service A deployed on environment QA and integration visible as Consul service.

```json
[
    {
        "Address": "10.32.108.37",
        "CreateIndex": 28,
        "ModifyIndex": 4992,
        "Node": "int02",
        "ServiceAddress": "10.32.108.37",
        "ServiceEnableTagOverride": false,
        "ServiceID": "int02:mesos-20160919:80",
        "ServiceName": "tracking",
        "ServicePort": 31440,
        "ServiceTags": [
            "integration"
        ],
        "TaggedAddresses": {
            "lan": "10.32.108.37",
            "wan": "10.32.108.37"
        }
    },
    {
        "Address": "10.32.108.33",
        "CreateIndex": 2463,
        "ModifyIndex": 5045,
        "Node": "test01",
        "ServiceAddress": "10.32.108.33",
        "ServiceEnableTagOverride": false,
        "ServiceID": "test01:mesos-20160919:80",
        "ServiceName": "tracking",
        "ServicePort": 31276,
        "ServiceTags": [
            "traefik.tags=testenv",
            "test",
            "traefik.frontend.rule=PathPrefix:/tracking;PathPrefixStrip:/tracking"
        ],
        "TaggedAddresses": {
            "lan": "10.32.108.33",
            "wan": "10.32.108.33"
        }
    }
]
```

You see both services got different tags. I set up my Traefik instance with 
```
[consulCatalog]
  endpoint = "127.0.0.1:8500"
  watch = true
  constraints = ["tag==testenv"]
```
and would expect just to see the second service instance in Traefik but I still see all my services known to consul including the first service shown here. 
Am I wrong? What would I have to do to make all services by an environment routable by just the corresponding Traefik instance on that environment? 

![screen shot 2016-09-29 at 15 58 32](https://cloud.githubusercontent.com/assets/8676888/18956963/9d3721c0-865d-11e6-80c8-c751b4c81b74.png)

**Comment by ummecasino** [comment]
Figured it out thanks to https://github.com/containous/traefik/issues/451, the solution is to put it like:

```
  [[consulCatalog.constraints]]
    key = "tag"
    mustmatch = true
    regex = "testenv"
```

**Comment by emilevauge** [comment]
@ummecasino constraints have been fixed in v1.1.0-rc1, you should now be able to write `constraints = ["tag==testenv"]` directly ;)
